com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -12,282 +14,180 @@ namespace MLAPI.Prototyping
-            SetupVar(m_NetworkRotation, m_CurrentRotation, ref m_OldRotation);
-            SetupVar(m_NetworkWorldScale, m_CurrentScale, ref m_OldScale);
+            if (Authority == NetworkAuthority.Client && IsClient && IsOwner)
+            {
+                // todo MTT-768 this shouldn't happen anymore with new tick system (tick written will be higher than tick read, so netvar wouldn't change in that case)
+                return;
+            }
 
-            UpdateVarPermissions();
+            ApplyNetworkState(newState);
         }
 
-        public override void OnNetworkObjectParentChanged(NetworkObject parentNetworkObject)
+        private void UpdateNetVarPerms()
         {
-            // TODO: handle parent NetworkObject change and potentially optimize pos/rot/scale replication?
+            switch (Authority)
+            {
+                default:
+                case NetworkAuthority.Server:
+                    m_NetworkState.Settings.WritePermission = NetworkVariablePermission.ServerOnly;
+                    break;
+                case NetworkAuthority.Client:
+                    m_NetworkState.Settings.WritePermission = NetworkVariablePermission.OwnerOnly;
+                    break;
+                case NetworkAuthority.Shared:
+                    m_NetworkState.Settings.WritePermission = NetworkVariablePermission.Everyone;
+                    break;
+            }
         }
 
-        private NetworkVariable<T>.OnValueChangedDelegate GetOnValueChangedDelegate<T>(Action<T> assignCurrent)
+        private void Awake()
         {
-            return (old, current) =>
-            {
-                if (TransformAuthority == Authority.Client && IsClient && IsOwner)
-                {
-                    // this should only happen for my own value changes.
-                    // todo MTT-768 this shouldn't happen anymore with new tick system (tick written will be higher than tick read, so netvar wouldn't change in that case
-                    return;
-                }
+            UpdateNetVarPerms();
 
-                assignCurrent.Invoke(current);
-            };
-        }
+            m_NetworkState.Settings.SendNetworkChannel = Channel;
+            m_NetworkState.Settings.SendTickrate = FixedSendsPerSecond;
 
-        private void Start()
-        {
-            // Register on value changed delegate. We can't simply check the position every fixed update because of shared authority
-            // Shared authority involves writing locally but applying changes when they come from the server. You can't both read from
-            // your NetworkPosition and write to it in the same FixedUpdate, you need both separate.
-            // There's no conflict resolution here. If two clients try to update the same value at the same time, they'll both think they are right
-            m_PositionChangedDelegate = GetOnValueChangedDelegate<Vector3>(current =>
```

Comment:
What happened to all of this? This was fixing a bug where with client authority on value changed gets invoked twice.

Reply:
we're still ignoring "loopback" onvaluechanged events, take a look at `OnNetworkStateChanged(...)` method, see the `if (Authority == NetworkAuthority.Client && IsClient && IsOwner)` check.